collectd/src/main/java/com/microsoft/applicationinsights/collectd/ApplicationInsightsWriter.java

```diff
@@ -0,0 +1,227 @@
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.collectd;
+
+import java.util.HashMap;
+import java.util.List;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.collectd.internal.ApplicationInsightsWriterLogger;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.internal.util.LocalStringsUtils;
+import com.microsoft.applicationinsights.telemetry.MetricTelemetry;
+import org.collectd.api.*;
+
+/**
+ * Created by yonisha on 4/26/2015.
+ */
+public class ApplicationInsightsWriter implements CollectdConfigInterface,
+        CollectdInitInterface,
+        CollectdWriteInterface,
+        CollectdShutdownInterface {
+
+    // region Consts
+
+    private static final String PLUGIN_NAME = "ApplicationInsightsWriter";
+    private static final String INSTRUMENTATION_KEY_CONFIGURATION_KEY = "InstrumentationKey";
+    private static final String SDK_LOGGER_CONFIGURATION_KEY = "SDKLogger";
+
+    // endregion Consts
+
+    // region Members
+
+    private TelemetryConfiguration telemetryConfiguration;
+    private TelemetryClient telemetryClient;
+    private ApplicationInsightsWriterLogger logger = null;
+
+    // endregion Members
+
+    // region Ctors
+
+    public ApplicationInsightsWriter() {
+        Collectd.registerConfig(PLUGIN_NAME, this);
+        Collectd.registerInit(PLUGIN_NAME, this);
+        Collectd.registerWrite(PLUGIN_NAME, this);
+        Collectd.registerShutdown(PLUGIN_NAME, this);
+
+        this.logger = new ApplicationInsightsWriterLogger();
+        this.telemetryConfiguration = TelemetryConfiguration.getActive();
+    }
+
+    protected ApplicationInsightsWriter(TelemetryClient telemetryClient, ApplicationInsightsWriterLogger logger) {
```

Comment:
Who is going to use this constructor except of the test?
It this is it's only purpose then it would be cleaner to create a test class inheriting from this class and overriding the constructor.